src/app/components/core/notifications/notifications.component.ts

```diff
@@ -0,0 +1,237 @@
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { sortBy } from 'lodash';
+import { map } from 'rxjs/operators';
+
+import {
+  AlertService, NotificationsService, ProgressBarService, SchedulesService, ServicesHealthService
+} from '../../../services';
+import { AlertDialogComponent } from '../../common/alert-dialog/alert-dialog.component';
+import { NotificationModalComponent } from './notification-modal/notification-modal.component';
+
+@Component({
+  selector: 'app-notifications',
+  templateUrl: './notifications.component.html',
+  styleUrls: ['./notifications.component.css'],
+  providers: [ServicesHealthService]
+})
+export class NotificationsComponent implements OnInit {
+
+  isNotificationServiceAvailable = false;
+  isNotificationServiceEnabled = false;
+  notificationServiceName = 'FogLAMP Notifications';
+  notificationInstances = [];
+  notification: any;
+  public notificationServiceRecord: any;
+
+  public showSpinner = false;
+  @ViewChild(NotificationModalComponent) notificationModal: NotificationModalComponent;
+  @ViewChild(AlertDialogComponent) child: AlertDialogComponent;
+
+  constructor(public servicesHealthService: ServicesHealthService,
+    public schedulesService: SchedulesService,
+    public notificationService: NotificationsService,
+    public ngProgress: ProgressBarService,
+    public alertService: AlertService,
+    private route: ActivatedRoute,
+    public router: Router) { }
+
+  ngOnInit() {
+    this.route.data.pipe(map(data => data['service'].services))
+      .subscribe(res => {
+        const serviceFound = res.some((service: any) => service.type === 'Notification' && service.status === 'running');
+        if (serviceFound) {
+          this.isNotificationServiceAvailable = true;
+          this.isNotificationServiceEnabled = true;
+        } else {
+          this.getSchedules();
+        }
+      },
+        (error) => {
+          if (error.status === 0) {
+            console.log('service down ', error);
+          } else {
+            this.alertService.error(error.statusText);
+          }
+        });
+    this.getNotificationInstance();
+  }
+
+  addNotificationService() {
+    const payload = {
+      name: this.notificationServiceName,
+      type: 'notification',
+      enabled: true
+    };
+
+    /** request start */
+    this.ngProgress.start();
+
+    this.servicesHealthService.addService(payload)
+      .subscribe(
+        () => {
+          /** request done */
+          this.ngProgress.done();
+          this.alertService.success('Notification service added successfully.', true);
```

Comment:
what do we get from Server API?

Reply:
return from server
`{"name": "FogLAMP Notifications", "id": "e8064fb9-e302-41b6-8a02-a290c41305b7"}`

Reply:
Okay! then its fine. May be we would have alert `.success('<name> service added successfully.', `